### PR TITLE
fix: app links reset network

### PR DESCRIPTION
**Pull request by DanielSchiavini**
- Add the current network to the cross-app urls
- Add tests

### CODE REVIEW

File: apps/lend/src/layout/Header.tsx [hunk 1]
```diff
@@ -32,7 +32,7 @@ const Header = ({ chainId, sections }: HeaderProps) => {
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [footerHeight])
 
-  const { rLocalePathname } = getParamsFromUrl()
+  const { rLocalePathname, rNetwork } = getParamsFromUrl()
```
Comment (0xAlunara) [inline apps/lend/src/layout/Header.tsx:35]: You're using `getNetworkFromUrl()` in `dao` and `loan` apps, but `getParamsFromUrl()` in `lend` and `main` apps. I was expecting it to be the same for all 4. Probably still works and might be something to fix once we harmonize all apps into a single layout.

EDIT: ah nvm I see `getParamsFromUrl()` calls `getNetworkFromUrl()` and some more.

File: apps/main/src/layout/default/Header.tsx [hunk 2]
```diff
@@ -57,7 +58,7 @@ export const Header = ({ sections }: HeaderProps) => {
         () =>
           _parseRouteAndIsActive(
             [
-              ...(hasRouter && (!network || network?.showRouterSwap)
+              ...(hasRouter && (!network || network.showRouterSwap)
```
Comment (0xAlunara) [inline apps/main/src/layout/default/Header.tsx:61]: Still puzzles me how the linter doesn't catch these needless null checks.

File: apps/main/src/layout/default/Header.tsx [hunk 3]
```diff
@@ -69,7 +70,7 @@ export const Header = ({ sections }: HeaderProps) => {
             ],
             rLocalePathname,
             routerPathname,
-            routerNetwork,
+            rNetwork,
           ),
         [hasRouter, network, networks, rChainId, rLocalePathname, routerCached, routerNetwork, routerPathname],
```
Comment (0xAlunara) [inline apps/main/src/layout/default/Header.tsx:75]: It's a bit confusing how there's now both `rNetwork` from the url and `routerNetwork` also from the url. Can this be harmonized / simplified?

**Review by 0xAlunara — COMMENTED** [review]
Really can't wait to merge the apps so there's hopefully less router state hacks required.

It works when navigating between `dex` and `lend` app, but once you go to `crvusd` or `dao` it basically loses whatever the network was and defaults back to `ethereum`, probably because these only live on mainnet and there's no network switcher there.

If that is acceptable that's fine, although I'm not sure that was the intent given you've also changed Header.tsx in both the `dao` and `loan` app.

File: apps/main/src/layout/default/Header.tsx (hunk 3 above)
Comment (DanielSchiavini) [reply apps/main/src/layout/default/Header.tsx:75]: This actually was not working with routerNetwork

It is something to dig into

File: apps/main/src/layout/default/Header.tsx (hunk 2 above)
Comment (DanielSchiavini) [reply apps/main/src/layout/default/Header.tsx:61]: Well this was not bad just redundant

**Review by 0xAlunara — APPROVED** [review]

**Pull request merged**